packages/frontend/src/App.tsx

```diff
@@ -80,6 +79,7 @@ function App() {
 
   const handleLanguageSelect = (language: string) => {
     setSelectedLanguage(language);
+    setCurrentEmojis([]); // these are per-language, so we have to clear them too
```

Comment:
this is a bit of a visual regression -- if we don't reset it, then switching languages will initially reuse the previous language's data (which is incorrect) and then every cell will flash when we receive the ACTUAL data (because each cell's count "changed"). 

BUT resetting it means that when we switch tabs, initially it'll be blank, so it looks a bit worse. maybe it'd make sense to use some kinda fallback here